File: tools/testing/testparser/rusttest.go
// Copyright 2020 The Fuchsia Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

package testparser

import (
	"fmt"
	"regexp"
)

var (
	rustTestPreamblePattern = regexp.MustCompile(`^running \d* tests?$`)
	rustTestCasePattern     = regexp.MustCompile(`^test (?:(.*?)::)?(.*?) \.\.\. (\w*)$`)
)

func parseRustTest(lines [][]byte) []TestCaseResult {
	var res []TestCaseResult
	for _, line := range lines {
		line := string(line)
		m := rustTestCasePattern.FindStringSubmatch(line)
		if m == nil {
			continue
		}
		suiteName := m[1]
		caseName := m[2]
		displayName := caseName
		if suiteName != "" {
			displayName = fmt.Sprintf("%s::%s", suiteName, caseName)
		}
		var status TestCaseStatus
		switch m[3] {
		case "ok":
			status = Pass
		case "FAILED":
			status = Fail
		case "ignored":
			status = Skip
		}
		res = append(res, TestCaseResult{
			DisplayName: displayName,
			SuiteName:   suiteName,
			CaseName:    caseName,
			Status:      status,
			Format:      "Rust",
		})
	}
	return res
}

File: src/devices/tests/bindlib-codegen-test/rust_test.rs
// Copyright 2022 The Fuchsia Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

use anyhow::Result;
use bind_bindlib_codegen_testlib;
use bind_bindlib_codegen_testlib::bind_fuchsia_pci;
use fuchsia_async as fasync;

#[fasync::run_singlethreaded(test)]
async fn test_constants_accessible() -> Result<()> {
    assert_eq!("bindlib.codegen.testlib.kinglet", bind_bindlib_codegen_testlib::KINGLET);
    assert_eq!(3839, bind_fuchsia_pci::BIND_PCI_VID_TEST);
    Ok(())
}
